### PR TITLE
Bugfix/rework: IPv6 scope errors (bp)

### PR DESCRIPTION
**Pull request by isbm**
One-to-one (almost) backport of bugfix, available at https://github.com/saltstack/salt/pull/49705

### CODE REVIEW

**Review by terminalmage — APPROVED** [review]

**Review by gtmanfred — APPROVED** [review]

File: tests/unit/modules/test_network.py [hunk 1]
```diff
@@ -276,7 +267,7 @@ def test_connect(self):
                     self.assertDictEqual(network.connect('host', 'port'),
                                          {'comment': ret, 'result': True})
 
-    @skipIf(HAS_IPADDRESS is False, 'unable to import \'ipaddress\'')
+    @skipIf(bool(ipaddress), 'unable to import \'ipaddress\'')
```
Comment (DmitryKuzmenko) [inline tests/unit/modules/test_network.py]: `@skipIf(not ipaddress, ...`?

File: tests/unit/modules/test_network.py [hunk 2]
```diff
@@ -288,7 +279,7 @@ def test_is_private(self):
                           return_value=True):
             self.assertTrue(network.is_private('::1'))
 
-    @skipIf(HAS_IPADDRESS is False, 'unable to import \'ipaddress\'')
+    @skipIf(bool(ipaddress), 'unable to import \'ipaddress\'')
```
Comment (DmitryKuzmenko) [inline tests/unit/modules/test_network.py]: `@skipIf(not ipaddress, ...`?

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
Looks good to me except the `skipIf` in tests, it looks you've reverted the bool logic.

**Review by DmitryKuzmenko — APPROVED** [review]
:+1:

**Review by meaksh — APPROVED** [review]

**Pull request merged**